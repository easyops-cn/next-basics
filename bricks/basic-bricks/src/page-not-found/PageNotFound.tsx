import React from "react";
import { useTranslation } from "react-i18next";
import { NS_BASIC_BRICKS, K } from "../i18n/constants";

interface PageNotFoundProps {
  url: string;
}

export function PageNotFound(props: PageNotFoundProps): React.ReactElement {
  const { t } = useTranslation(NS_BASIC_BRICKS);

  return (
    <div style={{ textAlign: "center", marginTop: 100 }}>
      <p style={{ fontSize: "150%" }}>{t(K.PAGE_NOT_FOUND_TITLE)}</p>
      <p>
        {t(K.PAGE_NOT_FOUND_DESC, {
          url: props.url
        })}
      </p>
    </div>
  );
}
