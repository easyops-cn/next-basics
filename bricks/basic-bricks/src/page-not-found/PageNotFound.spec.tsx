import React from "react";
import { shallow } from "enzyme";
import { PageNotFound } from "./PageNotFound";

describe("PageNotFound", () => {
  it("should work", () => {
    const wrapper = shallow(<PageNotFound url="http://example.com" />);
    expect(wrapper).toMatchSnapshot();
  });
});
