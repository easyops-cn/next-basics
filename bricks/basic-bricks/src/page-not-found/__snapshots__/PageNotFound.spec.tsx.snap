// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound should work 1`] = `
<div
  style={
    Object {
      "marginTop": 100,
      "textAlign": "center",
    }
  }
>
  <p
    style={
      Object {
        "fontSize": "150%",
      }
    }
  >
    PAGE_NOT_FOUND_TITLE
  </p>
  <p>
    PAGE_NOT_FOUND_DESC
  </p>
</div>
`;
